core/src/main/java/com/graphhopper/routing/util/BikeCommonFlagEncoder.java

```diff
@@ -586,11 +571,20 @@ void collect(IntsRef edgeFlags, ReaderWay way, double wayTypeSpeed,
     }
 
     protected void handleAccess(IntsRef edgeFlags, ReaderWay way) {
-
-        if (isOneway(way) || roundaboutEnc.getBool(false, edgeFlags)) {
-            boolean isBackward = isBackwardOneway(way);
-            accessEnc.setBool(isBackward, edgeFlags, true);
+        boolean isRoundabout = roundaboutEnc.getBool(false, edgeFlags);
+        boolean isOnewayForCars = isRoundabout || way.hasTag("oneway", oneways);
+        boolean isBackwardOnewayForCars = way.hasTag("oneway", "-1");
+        boolean hasBikeExceptionToOneway = (
+            way.hasTag("oneway:bicycle", "no")
+            || way.hasTag("cycleway", oppositeLanes)
```

Comment:
what does `oppositeLanes` mean here? maybe it's worth a comment explaining every exception?